pkg/proxy/request.go

```diff
@@ -200,17 +205,26 @@ func getEnvVars(r *http.Request) (map[string]string, error) {
 func getTestData(r *http.Request) (string, error) {
 	stringTestData, fileType, err := getFileFromHTTP(r, testData)
 	if err != nil {
+		//this means there was no testData file specified and we can ignore this error because testData is optional
+		if err == http.ErrMissingFile {
+			return "", nil
+		}
 		return "", err
 	}
 
-	if fileType == "csv" {
-		err = checkCsvFile(stringTestData)
-		if err != nil {
-			return "", err
+	for _, f := range testDataFileFormats {
+		if fileType == f {
+			if fileType == "csv" {
+				err = checkCsvFile(stringTestData)
+				if err != nil {
+					return "", err
+				}
+			}
+			return stringTestData, nil
```

Comment:
You could change `testDataFileFormats` to be a map of format: checker (`map[string]func(string) (string,error)`) that way the loop and nested ifs are avoided, looking something like this in the end:

```golang
if formatCheck, acceptedFormat := testDataFileFormats[fileType]; acceptedFormat {
    return formatCheck(stringTestData)
}
return "", ErrWrongFileFormat
```

and the map being:
```golang
type formatCheckFunc func (string) (string,error)

var testDataFileFormats = map[string]formatCheckFunc{
    "csv" : func (s string) (string, error) {
        if err = checkCsvFile(stringTestData); err != nil {
            return "", err
        }
        return s, nil
    },
    "protoset": func (s string) (string,error) { return s, nil }
}

Reply:
Following your suggestion below, I prefer to represent `testDataFileFormats` as map, the same way as `testFileFormats`, but not to add a `formatCheckFunc` inside the map. I think suggested code is a bit more difficult to read than a separate check applied only to csv files.